core/test/routing.spec.js

```diff
@@ -546,5 +547,92 @@ describe('Routing', () => {
       component.set({ viewUrl: 'http://otherurl.de/app.html!#/someUrl' });
       assert.isTrue(routing.isNotSameDomain(config, component));
     });
+
```

Comment:
We could refactor this tests to remove duplication of the mock data. The following approach should be fine, since each test case is receiving a fresh mock data object, so mutation in one test case does not affect other test cases:
```
    describe('defaultPathSegments', () => {
      const getDefaultPathSegment = routing.__get__('getDefaultPathSegment');
      const getBasePathData = function() {
        return {
          navigationPath: [
            {
              // DOESN'T MATTER
            },
            {
              pathSegment: 'groups',
              children: [
                {
                  pathSegment: 'stakeholders',
                  viewUrl:
                    '/sampleapp.html#/projects/1/users/groups/stakeholders'
                },
                {
                  pathSegment: 'customers',
                  viewUrl: '/sampleapp.html#/projects/1/users/groups/customers'
                }
              ]
            }
          ],
          context: {}
        }
      }

      it('should return first child if no dps set', () => {
        const pathData = getBasePathData();
        assert.equal(getDefaultPathSegment(pathData), 'stakeholders');
      });

      it('should child with pathSegment equal to defaultPathSegment', () => {
        const pathData = getBasePathData();
        pathData.navigationPath[1].defaultPathSegment = 'customers';
        assert.equal(getDefaultPathSegment(pathData), 'customers');
      });

      it('should return first child if given defaultPathSegment does not exist', () => {
        const pathData = getBasePathData();
        pathData.navigationPath[1].defaultPathSegment = 'NOSUCHPATH';
        assert.equal(getDefaultPathSegment(pathData), 'stakeholders');
      });
    });
```

Additionally, please also remove the describe block in line 447 if you agree it is unnecessary.